pkg/sfu/buffer/rtpstats_sender.go

```diff
@@ -647,8 +649,17 @@ func (r *RTPStatsSender) GetRtcpSenderReport(ssrc uint32, publisherSRData *RTCPS
 
 	timeSincePublisherSR := time.Since(publisherSRData.AtAdjusted)
 	now := publisherSRData.AtAdjusted.Add(timeSincePublisherSR)
-	nowNTP := mediatransportutil.ToNtpTime(now)
-	nowRTPExt := publisherSRData.RTPTimestampExt - tsOffset + uint64(timeSincePublisherSR.Nanoseconds()*int64(r.params.ClockRate)/1e9)
+	var (
+		nowNTP    mediatransportutil.NtpTime
+		nowRTPExt uint64
+	)
+	if cPassthroughNTPTimestamp {
+		nowNTP = publisherSRData.NTPTimestamp
+		nowRTPExt = publisherSRData.RTPTimestampExt - tsOffset
+	} else {
+		nowNTP = mediatransportutil.ToNtpTime(now)
+		nowRTPExt = publisherSRData.RTPTimestampExt - tsOffset + uint64(timeSincePublisherSR.Nanoseconds()*int64(r.params.ClockRate)/1e9)
```

Comment:
This is adjusting the timestamp to current wall clock time.

This is not possible when time stamps are passed through as is. So, what gets passed through may be late. Should be okay. They are like late arriving reports.

Being late is another issue. Have seen cases where the `timeSincePublisherSR` could be quite high (of the order of minutes), but I think they might all be bad client behaviour.

The better way to implement pass through would be make it fully event driven where reception of RTCP sender report from publisher triggers a send down stream. But, that will sacrifice efficiency in terms of more RTCP packets down stream (compound packets will not happen).

Another option is to make it event driven and batch it once a second or so to get both responsiveness and compound packet efficiency.